packages/wdio-logger/src/node.ts

```diff
@@ -112,7 +112,11 @@ const wdioLoggerMethodFactory = function (this: log.Logger, methodName: log.LogL
                 logCache.clear()
             }
 
-            return logFile.write(logText)
+            if (!logsContainInitPackageError(logText)) {
+                return logFile.write(logText)
+            }
+            // If we get Error during init of integration packages, write logs to both "outputDir" and the terminal
+            logFile.write(logText)
```

Comment:
This seems to do the same thing `logFile.write(logText)` or am I missing something?

Reply:
Yes, that's correct, the expression `logFile.write(logText)` is used twice. Here it is necessary to cover different scenarios.

In the first case, when the logs do not contain the target error related to the inability to find the integration package, everything happens as usual - this statement is used `return logFile.write(logText)`, logs are made to the file, and the flow ends with `return`
<details><summary>Case 1</summary>
<p>

```typescript
            if (!logsContainInitPackageError(logText)) {
                return logFile.write(logText) // << write to file
            }
            logFile.write(logText) // << this method is not called
``` 
</p>
</details> 


In the second case, when the logs contain the target error, logs are also made to the file, but the flow does not stop at this point; instead, it continues further. Additionally, the error is output to the terminal using statement `rawMethod(...args)`
<details><summary>Case 2</summary>
<p>

```typescript
            if (!logsContainInitPackageError(logText)) {
                return logFile.write(logText) // << not called
            }
            logFile.write(logText) // << write error to file when 'outputDir' is specified
        }
        ....
        rawMethod(...args) // << also write error to the terminal
``` 
</p>
</details>

Reply:
Why do we need the if statement then? Am I missing something?